zcash_primitives/src/transaction/builder.rs

```diff
@@ -405,6 +483,10 @@ impl<'a, P: consensus::Parameters, R: RngCore> Builder<'a, P, R> {
             transparent_bundle,
             sprout_bundle: None,
             sapling_bundle,
+            // This is not of the right type, and it's not obvious how to implement this generically
+            // for both a `Bundle<InProgress<Unproven, Unauthorized>` and a `Bundle<Authorized, V>`
+            // As such, it's created and authorized at the same time, right before being added to the
+            // authorized bundle
```

Comment:
It might be useful to look at https://github.com/zcash/zcash/blob/master/src/rust/src/builder_ffi.rs to see how the builder is being used inside of zcashd. It is likely that the `build_internal` method should be split into two different operations and the appropriate changes be made to the Sapling and transparent builders to support that separation.

Reply:
Thanks, I'll look there next!